### PR TITLE
deleted iam policy docs and references to iam policy in other docs

### PR DESCRIPTION
**Pull request by jc-berger**
See Jira: https://issues.redhat.com/browse/ACM-11650 

See PR for changes in the release notes for iam policy: https://github.com/stolostron/rhacm-docs/pull/6501

### CODE REVIEW

File: clusters/hosted_control_planes/deploy_aws_private_clusters.adoc [hunk 1]
```diff
@@ -19,85 +18,6 @@ For private clusters on AWS, all communication with the guest cluster occurs ove
 
 To enable private hosted clusters for AWS, you must first enable AWS PrivateLink. For more information, see xref:../../clusters/hosted_control_planes/enable_aws_private_link.adoc#hosted-enable-private-link[Enabling AWS PrivateLink].
 
-[#create-aws-private-hosted-cluster]
```
Comment (mprahl) [inline clusters/hosted_control_planes/deploy_aws_private_clusters.adoc]: This is a different IAM Policy and not related to the ACM IamPolicy kind that we removed. I would keep this section.

File: observability/observability_enable.adoc [hunk 2]
```diff
@@ -213,7 +213,6 @@ For Amazon S3 or S3 compatible storage, you can also use short term, limited-pri
 
 Generating access keys using AWS Security Service require the following additional steps:
 
-. Create an IAM policy that limits access to an S3 bucket.
```
Comment (mprahl) [inline observability/observability_enable.adoc]: This looks to be related to IAM in AWS and not IamPolicy so I'd keep this.

File: clusters/hosted_control_planes/deploy_aws_private_clusters.adoc (hunk 1 above)
Comment (dockerymick) [reply clusters/hosted_control_planes/deploy_aws_private_clusters.adoc]: @jc-berger

File: clusters/hosted_control_planes/deploy_aws_private_clusters.adoc [hunk 3]
```diff
@@ -10,7 +10,6 @@ For private clusters on AWS, all communication with the guest cluster occurs ove
 *Important:* Although public clusters can be created in any region, private clusters can be created only in the region that is specified by `--aws-private-region`.
 
 * <<prerequisites-aws-private-clusters,Prerequisites>>
-* <<create-aws-private-hosted-cluster,Creating a private hosted cluster on AWS>>
```
Comment (mprahl) [inline clusters/hosted_control_planes/deploy_aws_private_clusters.adoc]: This file should be left as is since it's not policy related.

Comment (dockerymick) [reply clusters/hosted_control_planes/deploy_aws_private_clusters.adoc]: @jc-berger reminder to add this child link back to the file

File: clusters/hosted_control_planes/deploy_aws_private_clusters.adoc [hunk 4]
```diff
@@ -96,7 +95,7 @@ hcp create cluster aws \
 The API endpoints for the cluster are accessible through a private DNS zone:
 
 - `api.<hosted-cluster-name>.hypershift.local`
-- `*.apps.<hosted-cluster-name>.hypershift.local`
```
Comment (dockerymick) [inline clusters/hosted_control_planes/deploy_aws_private_clusters.adoc]: I'm not sure why this was changed. I recommend that you add the asterisks back

**Review by dockerymick — CHANGES_REQUESTED** [review]

**Review by dockerymick — APPROVED** [review]

**Pull request merged**